usb: sam0: implement missing API functions for tests/subsys/usb/device

benpicco opened the pull request:
When pushing #13494 I noticed CI would fail for `tests/subsys/usb/device`.

It turns out the test case fails to build for sam0 as the usb driver for the platform does not implement three API functions:

 - `usb_dc_ep_disable()`
 - `usb_dc_ep_halt()`
 - `usb_dc_ep_flush()`

While *halt* and *disable* are trivial, I was unsure what to *flush* and the function seems optional in most cases, so I left it as an exercise for a later user :wink: 

After getting the test case compile again, I noticed it would always hard fault.
This is because the test case exercises the API with invalid endpoint indices and expects the driver to catch the error. This meant inserting checks whether the endpoint is withing the valid range into every function.

While I find this a bit excessive, the test case now runs successful again.

finikorg reviewed (approved):
LGTM (I would make 2 separate patches though)

benpicco committed:
usb: sam0: Ensure API arguments are valid

Make sure the parameters for the API functions are valid, return error
otherwise.

This is expected by the tests/subsys/usb/device test case.

Signed-off-by: Benjamin Valentin <benpicco@googlemail.com>

benpicco committed:
usb: sam0: Implement missing API functions

This implements three API functions that are required for
tests/subsys/usb/device to build:

 - usb_dc_ep_disable()
 - usb_dc_ep_halt()
 - usb_dc_ep_flush()

While halt and disable are trivial, flush is just a stub for now.

Signed-off-by: Benjamin Valentin <benpicco@googlemail.com>

Sizurka commented on drivers/usb/device/usb_dc_sam0.c:
```diff
@@ -424,6 +471,23 @@ int usb_dc_ep_enable(const u8_t ep)
+		return -1;
+	}
+
+	endpoint->EPINTENSET.reg = 0;
```
Shouldn't this be:
```suggestion
	endpoint->EPINTENCLR.reg = USB_DEVICE_EPINTENCLR_TRCPT0 |
					   USB_DEVICE_EPINTENCLR_TRCPT1 |
					   USB_DEVICE_EPINTENCLR_RXSTP;
```

Since writing 0 to `EPINTENSET` has (should have? I haven't tested) no effect.

Sizurka reviewed (approved):
Look good: test case passes with this applied.
